lang/cs/Org.Apache.REEF.IMRU/OnREEF/IMRUTasks/IMRUTaskAppException.cs

```diff
@@ -0,0 +1,42 @@
+﻿// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+using System;
+using System.Runtime.Serialization;
+
+namespace Org.Apache.REEF.IMRU.OnREEF.IMRUTasks
+{
+    /// <summary>
+    /// A serialiable exception that represents a task application error.
+    /// </summary>
+    [Serializable]
+    public sealed class IMRUTaskAppException : Exception
```

Comment:
I would say we should catch all the application errors at task lever then throw this type of exception, because at driver side, we only need to look at the type of the exception then make a decision from fault tolerant perspective. 
If we want to wrap any exceptions as inner exception, we need to look at if we are able to serialize those nested exceptions. Currently we only guarantee those few defined exceptions are serializable.